docs/run-preinstalled-wda.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Run Preinstalled WebDriverAgentRunner
+---
+
+XCUITest driver can launch preinstalled WebDriverAgent directly against a real device.
+It lets you start an Appium session without the `xcodebuild` command execution to improve the session start performance.
+
+`appium:webDriverAgentUrl` let you manage everything related to the WebDriverAgent, but this functionality let Appium to manage instead.
+
+## For Real Device
+
+### Capabilities
+
+- Required
+  - `appium:usePreinstalledWDA` to start a WebDriverAgent process without xcodebuild
+- Optional
+  - `appium:preInstalledWDABundleId` to customize the WebDriverAgent bundle id Appium launches
+
+### Example steps with Xcode
+
+1. Run WebDriverAgent with Xcode as test
+2. Stop the Xcode session
+3. Start an Appium session with the capabilities below:
+
+```ruby
+# Ruby
+capabilities: {
+  "platformName": "ios",
+  "appium:automationName": "xcuitest",
+  "appium:udid": "<udid>",
+  "appium:usePreinstalledWDA": true,
+  "appium:preInstalledWDABundleId": "com.appium.WebDriverAgentRunner.xctrunner"
+}
+@core = Appium::Core.for capabilities: capabilities
+@core.start_driver
+```
+
+If the `<udid>` device has a WebDriverAgent package with `com.appium.WebDriverAgentRunner.xctrunner` bundle id, the session will launch the WebDriverAgent process without xcodebuild.
+
+> **Note**
+> Please ensure that the bundle id is launchable before starting an Appium session.
+> For example, whether the provisioning profile is trusted by the system.
+
+### Install WebDriverAgent
+
+#### With Xcode
+
+Running test of [WebDriveragent](https://github.com/appium/WebDriverAgent) package with Xcode is the easiest way to prepare the device environment.
+Please read [Real Device Configuration tutorial](real-device-config.md) to configure the WebDriverAgent package for a real device.
+
+The `preInstalledWDABundleId` is usually `<bundle id>.xctrunner`. The `<bundle id>` could depend on your provisioning profile environment.
+If it is a non-paid account by `appium` user name, the bundle id would have `com.appium` prefix.
+Then, the WebDriverAgent bumdle id could be `com.appium.WebDriverAgentRunner` for example.
+The test bundle by Xcode will be `com.appium.WebDriverAegnt.xctrunner`.
+The `preInstalledWDABundleId` needs to be the `com.appium.WebDriverAegnt.xctrunner`.
+
+> **Note**
+> Please make sure if the installed WebDriverAgent is valid if the session start continued failing with a correct WebDriverAgent bundle id.
```

Comment:
Please make sure that the installed WebDriverAgent-Runner application is still launchable if the test session startup still fails by providing a correct WebDriverAgent bundle identifier.